staging/src/k8s.io/legacy-cloud-providers/azure/clients/fileclient/azure_fileclient.go

```diff
@@ -19,106 +19,73 @@ limitations under the License.
 package fileclient
 
 import (
+	"context"
 	"fmt"
-	"net/http"
 
-	azs "github.com/Azure/azure-sdk-for-go/storage"
-	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2019-06-01/storage"
 
 	"k8s.io/klog"
-	"k8s.io/legacy-cloud-providers/azure/retry"
+	azclients "k8s.io/legacy-cloud-providers/azure/clients"
 )
 
-const (
-	useHTTPS = true
-)
-
-var (
-	// refer https://github.com/Azure/azure-sdk-for-go/blob/master/storage/client.go#L88.
-	defaultValidStatusCodes = []int{
-		http.StatusRequestTimeout,      // 408
-		http.StatusInternalServerError, // 500
-		http.StatusBadGateway,          // 502
-		http.StatusServiceUnavailable,  // 503
-		http.StatusGatewayTimeout,      // 504
-	}
-)
-
-// AzureFileClient implements the azure file client interface
-type AzureFileClient struct {
-	env     *azure.Environment
-	backoff *retry.Backoff
+// Client implements the azure file client interface
+type Client struct {
+	fileSharesClient storage.FileSharesClient
 }
 
-// NewAzureFileClient creates a azure file client
-func NewAzureFileClient(env *azure.Environment, backoff *retry.Backoff) *AzureFileClient {
-	return &AzureFileClient{
-		env:     env,
-		backoff: backoff,
+// New creates a azure file client
+func New(config *azclients.ClientConfig) *Client {
+	client := storage.NewFileSharesClientWithBaseURI(config.ResourceManagerEndpoint, config.SubscriptionID)
+	client.Authorizer = config.Authorizer
+
+	return &Client{
+		fileSharesClient: client,
 	}
 }
 
 // CreateFileShare creates a file share
-func (f *AzureFileClient) CreateFileShare(accountName, accountKey, name string, sizeGiB int) error {
-	fileClient, err := f.getFileSvcClient(accountName, accountKey)
-	if err != nil {
-		return err
-	}
-	share := fileClient.GetShareReference(name)
-	share.Properties.Quota = sizeGiB
-	newlyCreated, err := share.CreateIfNotExists(nil)
-	if err != nil {
-		return fmt.Errorf("failed to create file share, err: %v", err)
+func (c *Client) CreateFileShare(resourceGroupName, accountName, name string, sizeGiB int) error {
+	quota := int32(sizeGiB)
+	fileShare := storage.FileShare{
+		Name: &name,
+		FileShareProperties: &storage.FileShareProperties{
+			ShareQuota: &quota,
+		},
 	}
-	if !newlyCreated {
-		klog.V(2).Infof("file share(%s) under account(%s) already exists", name, accountName)
-	}
-	return nil
+	_, err := c.fileSharesClient.Create(context.Background(), resourceGroupName, accountName, name, fileShare)
```

Comment:
https://docs.microsoft.com/en-us/rest/api/storageservices/create-share
```
The Create Share operation creates a new share under the specified account. If the share with the same name already exists, the operation fails.
```

I think if it return error, check error message, keep same behavior as old code.

Reply:
It is by design. The management plane API silent ignores the error and takes it as an update operation.

Reply:
make the logic consistent with the original. See https://github.com/kubernetes/kubernetes/pull/91723